### PR TITLE
Remove --credentials flags, stop generating router/registry client certs

### LINKED ISSUE #3951: openshift-node is logging private RSA keys to the systemd journal
```
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: ValueFrom:<nil>} {Name:OPENSHIFT_MASTER Value:https://losbs.example.com:8443 ValueFrom:<nil>} {Name:ROUTER_SERVICE_NAME Value:router ValueFrom:<nil>} {Name:ROUTER_SERVICE_NAMESPACE Value:default ValueFrom:<nil>} {Name:STATS_PASSWORD Value:BBVRF1e9gq ValueFrom:<nil>} {Name:STATS_PORT Value:1936 ValueFrom:<nil>} {Name:STATS_USERNAME Value:admin ValueFrom:<nil>}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:default-token-81jlw ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:0xc2096e20c0 ReadinessProbe:<nil> Lifecycle:<nil> TerminationMessagePath:/dev/termination-log ImagePullPolicy:IfNotPresent SecurityContext:0xc208d49a20}] RestartPolicy:Always TerminationGracePeriodSeconds:<nil> ActiveDeadlineSeconds:<nil> DNSPolicy:ClusterFirst NodeSelector:map[] ServiceAccountName:default NodeName:losbs.example.com HostNetwork:false ImagePullSecrets:[{Name:default-dockercfg-94xqt}]} Status:{Phase:Pending Conditions:[] Message: Reason: HostIP: PodIP: StartTime:<nil> ContainerStatuses:[]}}, podFullName: "router-1-hjuyt_default", uid: "80388a0e-357a-11e5-b69b-525400b16e9e"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.578517   18926 manager.go:1375] Found pod infra container for "router-1-hjuyt_default"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.579156   18926 manager.go:1364] Syncing Pod &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:docker-registry-1-i2bz5 GenerateName:docker-registry-1- Namespace:default SelfLink:/api/v1/namespaces/default/pods/docker-registry-1-i2bz5 UID:33b0d96c-3607-11e5-b69b-525400b16e9e ResourceVersion:7602 Generation:0 CreationTimestamp:2015-07-29 10:33:46 -0500 CDT DeletionTimestamp:<nil> Labels:map[deployment:docker-registry-1 deploymentconfig:docker-registry docker-registry:default] Annotations:map[kubernetes.io/created-by:{"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"default","name":"docker-registry-1","uid":"3238c5c1-3607-11e5-b69b-525400b16e9e","apiVersion":"v1","resourceVersion":"7600"}} openshift.io/deployment-config.latest-version:1 openshift.io/deployment-config.name:docker-registry openshift.io/deployment.name:docker-registry-1 openshift.io/scc:privileged kubernetes.io/config.source:api kubernetes.io/config.seen:2015-07-29T10:33:46.52852118-05:00]} Spec:{Volumes:[{Name:registry-storage VolumeSource:{HostPath:0xc208f1b320 EmptyDir:<nil> GCEPersistentDisk:<nil> AWSElasticBlockStore:<nil> GitRepo:<nil> Secret:<nil> NFS:<nil> ISCSI:<nil> Glusterfs:<nil> PersistentVolumeClaim:<nil> RBD:<nil> CephFS:<nil> Metadata:<nil>}} {Name:registry-token-kl6za VolumeSource:{HostPath:<nil> EmptyDir:<nil> GCEPersistentDisk:<nil> AWSElasticBlockStore:<nil> GitRepo:<nil> Secret:0xc208f1b330 NFS:<nil> ISCSI:<nil> Glusterfs:<nil> PersistentVolumeClaim:<nil> RBD:<nil> CephFS:<nil> Metadata:<nil>}}] Containers:[{Name:registry Image:openshift/origin-docker-registry:v1.0.3 Command:[] Args:[] WorkingDir: Ports:[{Name: HostPort:0 ContainerPort:5000 Protocol:TCP HostIP:}] Env:[{Name:OPENSHIFT_CA_DATA Value:-----BEGIN CERTIFICATE-----
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: MIIC5jCCAdCgAwIBAgIBATALBgkqhkiG9w0BAQswJjEkMCIGA1UEAwwbb3BlbnNo
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: aWZ0LXNpZ25lckAxNDM4MTIzMzMwMB4XDTE1MDcyODIyNDIwOVoXDTE2MDcyNzIy
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: NDIxMFowJjEkMCIGA1UEAwwbb3BlbnNoaWZ0LXNpZ25lckAxNDM4MTIzMzMwMIIB
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: IjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAujUr3rtMzXHnhWYNc/2dnEVy
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: VLD4UoDg/r4T4MnSPXwybPR6jTxQXeQiXAoYLBkIDYIohSBsFgl1MsFCba8ffRdB
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: mGag5v+J1+8Crv1C9vVCin/MYKxoO5Vvp0QIDSUtQVgUEnqyeSL9JmswwYtN3bI/
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 42/Ub+FoSFOZeF56azaYyUY5klo6e1zhPnhgfloTgiuwWCvAUmCF3pKOYblQIDOh
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 0NUZrk1Z17kVPdZ0ms8I4V+SG+PH0xvKftqlI+tjj9RhK7Ccy4qWTKyfOKdaUcnH
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: VpKNLgX8acvSC44XAm/PoSwbOwSm33xQg8HgbG4ki7mDwyVNsKxthHVzNbcX+wID
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: AQABoyMwITAOBgNVHQ8BAf8EBAMCAKQwDwYDVR0TAQH/BAUwAwEB/zALBgkqhkiG
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 9w0BAQsDggEBALbfYYZxnIy0cWkCRkODDLEfESvIhPlqPn+VmiEnJ53pk+ekHL1v
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: Zwn3k0u7I2TUmPdkATuEC0bdaCFsZAnDN7p0VaEVcgYfS92slCA3LEvThd9GzGay
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: z5L79APEpSz3u5Pa3fQfOOH+G+6SHlvjnzMysuZI4l6wne6AjhnjWpJv1vWLHdg2
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: ru3x+zoJlzCPlFyQvuT9zmNOOkjmOYW9xO8a/6fUT2c2+VtAA34eORNy9/1rQOi/
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: w1nCnY13w/5jZso1vNj7wnrxQddmTlZBQsv/OmXC0Is37Z9r/1OHSTZTOnirSSwv
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 7NxmGiQwbkrrzy5rSmdgNT+tnIV3aK9poHM=
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: -----END CERTIFICATE-----
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: ValueFrom:<nil>} {Name:OPENSHIFT_CERT_DATA Value:-----BEGIN CERTIFICATE-----
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: MIIDEjCCAfygAwIBAgIBBTALBgkqhkiG9w0BAQswJjEkMCIGA1UEAwwbb3BlbnNo
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: aWZ0LXNpZ25lckAxNDM4MTIzMzMwMB4XDTE1MDcyODIyNDIxNVoXDTE2MDcyNzIy
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: NDIxNlowQDEaMBgGA1UEChMRc3lzdGVtOnJlZ2lzdHJpZXMxIjAgBgNVBAMTGXN5
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: c3RlbTpvcGVuc2hpZnQtcmVnaXN0cnkwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAw
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: ggEKAoIBAQDLe7cpv3pBCd5sG/tJgyigHmIF+HpQpTVy2Ccs82xZM49Vii2Clwg4
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: DC0vvsBcaLSW8ngmF5JdjEDF2fVz46+cf1UKd2IENlIOqgcuc36gO1YHkAhYX6kY
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: NTA8Iztv4CbqzxRM86USwUGQSg/xvw8WEgBPEjg+K9S8BGB/xcihpLF2MbLbJxbj
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 7QMTTRKYZEd2ccd7CrXVOTwyWGa5nn4N/xfxneKPAEi6QI/vQRXP+SeDSSl0qEJ1
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: Fk08db9H0rMsxjtPccz1KggEMxgVGBC/zxc1kxkJCjzFuBwmvIcfQOxEjQDFnTTu
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 2VoaQ6Jd5GfVSGhS75UAgSfegAn1vmeVAgMBAAGjNTAzMA4GA1UdDwEB/wQEAwIA
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: oDATBgNVHSUEDDAKBggrBgEFBQcDAjAMBgNVHRMBAf8EAjAAMAsGCSqGSIb3DQEB
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: CwOCAQEAlw2dminQINeWQx96OCa7cpTfoX0RXWDkDFcYFc7xQvbGlkafodDocuB3
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: dp37IXRDRTboHA4vjP/6he/eiy5C+9pcQbnHnUSuUiwqs3xvbfZdjhFYEjqdPvP1
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: xhEjZZ4aYb7pUc3Dk88EZ/8IFIubh9g9grZHcHv94nfr21pL71X9EFBPatElTa2j
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: oOyraAyGRb1RN2RTvaABdWYtdtQRNRRsK0+mllMNeJptlMR/tKgON0YqwJndW4P4
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: Ijx2o2WR3Dkr0KoljW1YirVhEeqemVcehy96vYfphEWXV/Ezzo0TwbltoW014U6N
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: OrvyOr9Tyvu6gIWSG0SA6HoaS05Lxg==
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: -----END CERTIFICATE-----
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: ValueFrom:<nil>} {Name:OPENSHIFT_INSECURE Value:false ValueFrom:<nil>} {Name:OPENSHIFT_KEY_DATA Value:-----BEGIN RSA PRIVATE KEY-----
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: MIIEpgIBAAKCAQEAy3u3Kb96QQnebBv7SYMooB5iBfh6UKU1ctgnLPNsWTOPVYot
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: gpcIOAwtL77AXGi0lvJ4JheSXYxAxdn1c+OvnH9VCndiBDZSDqoHLnN+oDtWB5AI
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: WF+pGDUwPCM7b+Am6s8UTPOlEsFBkEoP8b8PFhIATxI4PivUvARgf8XIoaSxdjGy
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 2ycW4+0DE00SmGRHdnHHewq11Tk8MlhmuZ5+Df8X8Z3ijwBIukCP70EVz/kng0kp
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: dKhCdRZNPHW/R9KzLMY7T3HM9SoIBDMYFRgQv88XNZMZCQo8xbgcJryHH0DsRI0A
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: xZ007tlaGkOiXeRn1UhoUu+VAIEn3oAJ9b5nlQIDAQABAoIBAQCvreJXcu+UE3cr
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: E4WC1HBQvaqXyQ2hbuGUg7ZfAtM6uF851RCxvL/l0+13WVMu9V9lK3Y8PQGrLpoF
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: t8ZJBfzqOq0rWiMPy+5Qps+IdUKQCGqgIORzmDOfqNw6L5jzYC15cMvwHDMa3CtE
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 7gySBQh7X5NQsURJuQH0SjseVZ6AnuulWDMrlJ5HzIWZs2O4m2X7TEKc20M46Lm8
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: xci457HnqW8IaOLBgdJxGn1obWKvZ2wHVCm4RGqNL2uh3lSCrimez0xe/LZDxGqC
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: mo7n1H0osSp72dBN9RFg/VTX8pwJPS8GWQ1dA7RA1QSyAgETczCSFThM7Iqz1kYr
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 6N0HWxOJAoGBAPIdxpmLKKL9UzNcdf6U1OLsNKXsgOPCf5nyeSJ/57gYwbwQuYIk
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 5J3ImVvWiB6gnVNG3Po9q87D5hLo3j07jAzcEmQULmBEyzy7N24O6WA7oy1ymGky
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: qcjRevrp3wU8QsdO4QCk7iQfC2aCOJYhvOn8Kgoil2eZFsCV5YsedPh7AoGBANcm
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 0D6DXXKkH+ZA+ZDZ8th4YmdPHaREa+R65AKnlTCAadb4X8dxR/ri07QIml3Sdo2B
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: 0SxqsMFAVzvLsBeHoFllW1aHZ3zLLcjRtwEEh3jVZrmgW21SE6+RQCckDk7FR6yB
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: A6O63JyWGn9iaF0B1V1NhgU2EYtD/oFe2Y+77osvAoGBAKsl3zi0xSIKPuEksAgp
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: Px+15zWPLEFUQQ0a+QQpNCzP20tM60WTPMRIhBgGW6yCzxOKGTUPmT6M/jtJv6e7
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: SzHf82QpW+sTVjN1CCbVhaVQIvZCdniOprRdWDi8T7mMsovfN3BUJj8EbvOkg8mk
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: yb0BUtpdP3etIQc8Lsob2oW3AoGBAMJfxnMzqxEPReJ3gQBkZwk6eWxsBZOKXQ8m
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: FyhuzlHrMbOM4LsGlMkMc8aKnyt0ILbnOVqvaNCumCDlfElFtpANCd+6pLwRs2Ni
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: U7sPpHR+FzhMr0ysgixzRUoXQPXZNU+bnvYks3qkAXxkI3UjWufYSeQbXO48l4m5
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: b0RzvKbxAoGBAJ4zomgk25vzSvJ56pfQPwjJKC82QRcw9XTdnJB838Y8rCx9Yn+2
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: SXn1J/Nci8lT0fAs+aBj3M8F4hvQz/NB/MybBSuvqXskD3cDGAFH+2XQYsdrmbS8
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: NG6rK1GAKiLW9B8ntZ5Ouf+gDA9fhlLIYpXQb7Tz2vyZPjiOHHqlaOga
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: -----END RSA PRIVATE KEY-----
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: ValueFrom:<nil>} {Name:OPENSHIFT_MASTER Value:https://losbs.example.com:8443 ValueFrom:<nil>}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:registry-storage ReadOnly:false MountPath:/registry} {Name:registry-token-kl6za ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:<nil> ReadinessProbe:<nil> Lifecycle:<nil> TerminationMessagePath:/dev/termination-log ImagePullPolicy:IfNotPresent SecurityContext:0xc208d49aa0}] RestartPolicy:Always TerminationGracePeriodSeconds:<nil> ActiveDeadlineSeconds:<nil> DNSPolicy:ClusterFirst NodeSelector:map[] ServiceAccountName:registry NodeName:losbs.example.com HostNetwork:false ImagePullSecrets:[{Name:registry-dockercfg-ic68z}]} Status:{Phase:Pending Conditions:[] Message: Reason: HostIP: PodIP: StartTime:<nil> ContainerStatuses:[]}}, podFullName: "docker-registry-1-i2bz5_default", uid: "33b0d96c-3607-11e5-b69b-525400b16e9e"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.579252   18926 manager.go:1375] Found pod infra container for "docker-registry-1-i2bz5_default"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581288   18926 manager.go:1388] Pod infra container looks good, keep it "docker-registry-1-i2bz5_default"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581362   18926 manager.go:1411] pod "docker-registry-1-i2bz5_default" container "registry" exists as 1ae3f672c584c4706d6786b2aeb8fe3fc6cdbc2ff9b21b566b23e028b40e1dce
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581388   18926 prober.go:180] Readiness probe for "docker-registry-1-i2bz5_default:registry" succeeded
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581395   18926 manager.go:1442] probe success: "registry"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581408   18926 manager.go:1515] Got container changes for pod "docker-registry-1-i2bz5_default": {StartInfraContainer:false InfraContainerId:5a9a80933128aac8f54c5b9d9f1f18507eec67779155dbba14bcc0e104ef8619 ContainersToStart:map[] ContainersToKeep:map[5a9a80933128aac8f54c5b9d9f1f18507eec67779155dbba14bcc0e104ef8619:-1 1ae3f672c584c4706d6786b2aeb8fe3fc6cdbc2ff9b21b566b23e028b40e1dce:0]}
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581448   18926 kubelet.go:2245] Generating status for "docker-registry-1-i2bz5_default"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.581958   18926 manager.go:1388] Pod infra container looks good, keep it "router-1-hjuyt_default"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582063   18926 manager.go:1411] pod "router-1-hjuyt_default" container "router" exists as e7f81682f97116e4ecfda09d031102dc77c82427a9759460ed94320707572291
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582073   18926 prober.go:180] Readiness probe for "router-1-hjuyt_default:router" succeeded
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582080   18926 prober.go:225] TCP-Probe PodIP: 10.1.0.3, Port: 80, Timeout: 1s
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582662   18926 prober.go:136] Liveness probe for "router-1-hjuyt_default:router" succeeded
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582676   18926 manager.go:1442] probe success: "router"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582687   18926 manager.go:1515] Got container changes for pod "router-1-hjuyt_default": {StartInfraContainer:false InfraContainerId:c3c50b6ad474a5c19abd304a254e334a31daa4661c892046e06eaf56b5148955 ContainersToStart:map[] ContainersToKeep:map[c3c50b6ad474a5c19abd304a254e334a31daa4661c892046e06eaf56b5148955:-1 e7f81682f97116e4ecfda09d031102dc77c82427a9759460ed94320707572291:0]}
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.582715   18926 kubelet.go:2245] Generating status for "router-1-hjuyt_default"
Jul 29 15:25:09 losbs.example.com openshift-node[18926]: I0729 15:25:09.612163   18926 manager.go:279] Container inspect result: {ID:e7f81682f97116e4ecfda09d031102dc77c82427a9759460ed94320707572291 Created:2015-07-29 16:45:33.605714603 +0000 UTC Path:/usr/bin/openshift-router Args:[--template=/var/lib/haproxy/conf/haproxy-config.template --reload=/var/lib/haproxy/reload-haproxy] Config:0xc208496480 State:{Running:true Paused:false Restarting:false OOMKilled:false Pid:37041 ExitCode:0 Error: StartedAt:2015-07-29 16:45:34.34976607 +0000 UTC FinishedAt:0001-01-01 00:00:00 +0000 UTC} Image:6476e03a99d277b80c0ca36f813a5273a9658a078e7f92e32a7648c9847b44f8 Node:<nil> NetworkSettings:0xc209e0a230 SysInitPath: ResolvConfPath:/var/lib/docker/containers/c3c50b6ad474a5c19abd304a254e334a31daa4661c892046e06eaf56b5148955/resolv.conf HostnamePath:/var/lib/docker/containers/c3c50b6ad474a5c19abd304a254e334a31daa4661c892046e06eaf56b5148955/hostname HostsPath:/var/lib/docker/containers/c3c50b6ad474a5c19abd304a254e334a31daa4661c892046e06eaf56b5148955/hosts Name:/k8s_router.e0da39d5_router-1-hjuyt_default_80388a0e-357a-11e5-b69b-525400b16e9e_5aeca6e0 Driver:devicemapper Volumes:map[/dev/termination-log:/var/lib/openshift/openshift.local.volumes/pods/80388a0e-357a-11e5-b69b-525400b16e9e/containers/router/e7f81682f97116e4ecfda09d031102dc77c82427a9759460ed94320707572291 /var/run/secrets/kubernetes.io/serviceaccount:/var/lib/openshift/openshift.local.volumes/pods/80388a0e-357a-11e5-b69b-525400b16e9e/volumes/kubernetes.io~secret/default-token-81jlw] VolumesRW:map[/dev/termination-log:true /var/run/secrets/kubernetes.io/serviceaccount:false] HostConfig:0xc2085b41e0 ExecIDs:[] AppArmorProfile:}
```

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #3951

doc in https://github.com/openshift/openshift-docs/pull/3552

ansible hasn't used this since 3.2

### CODE REVIEW

**Comment by liggitt** [comment]
post-1.3

**Comment by liggitt** [comment]
[test]

**Comment by kseifriedredhat** [comment]
Do we have an ETA on when this will be cherry picked into OpenShift Enterprise 3.x? Then I'll attach CVE-2015-8945 to the errata in question and we can close this out.

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
@deads2k @sdodson PTAL

**Comment by sdodson** [comment]
I don't see any problems with this and our current installer code.

File: pkg/cmd/admin/registry/registry.go [hunk 1]
```diff
@@ -294,36 +284,11 @@ func (opts *RegistryOptions) RunCmdRegistry() error {
...
-			"OPENSHIFT_KEY_DATA":  string(config.KeyData),
-			"OPENSHIFT_CERT_DATA": string(config.CertData),
-			"OPENSHIFT_INSECURE":  fmt.Sprintf("%t", config.Insecure),
-		}
+	if len(opts.Config.ServiceAccount) == 0 {
+		return fmt.Errorf("registry could not be created; a service account must be provided")
 	}
 
-	needServiceAccountRole := len(opts.Config.ServiceAccount) > 0 && len(opts.Config.Credentials) == 0
+	needServiceAccountRole := len(opts.Config.ServiceAccount) > 0
```
Comment (deads2k) [inline pkg/cmd/admin/registry/registry.go]: Didn't you just make sure they specified it? this bool seems unnecessary.

File: pkg/cmd/admin/router/router.go [hunk 2]
```diff
@@ -619,35 +606,10 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
...
-			"OPENSHIFT_CA_DATA":   string(config.CAData),
-			"OPENSHIFT_KEY_DATA":  string(config.KeyData),
-			"OPENSHIFT_CERT_DATA": string(config.CertData),
-			"OPENSHIFT_INSECURE":  insecure,
-		})
+	if len(cfg.ServiceAccount) == 0 {
+		return fmt.Errorf("router could not be created; you must specify a service account with --service-account")
 	}
-	createServiceAccount := len(cfg.ServiceAccount) > 0 && len(cfg.Credentials) == 0
+	createServiceAccount := len(cfg.ServiceAccount) > 0
```
Comment (deads2k) [inline pkg/cmd/admin/router/router.go]: always true?

**Comment by deads2k** [comment]
minor comments. lgtm otherwise.

File: pkg/cmd/admin/router/router.go (hunk 2 above)
Comment (liggitt) [inline pkg/cmd/admin/router/router.go]: yes, removed

File: pkg/cmd/admin/registry/registry.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/admin/registry/registry.go]: yes

**Comment by liggitt** [comment]
[merge]

**Pull request merged**